packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -271,6 +271,35 @@ abstract class ProcessUtils {
     );
     return completer.future;
   }
+
+  /// See [writelnToStdinGuarded]. This calls `stdin.write` instead of `stdin.writeln`.
+  static Future<void> writeToStdinGuarded({
```

Comment:
should we just have `writelnToStdinGuarded` take an optional bool of whether to add a newline at the end? In which case it would make sense to rename it to writeToStdinGuarded.

Reply:
My goal here was to preserve the user ergonomics of `IOSink` by having separate methods for `write` and `writeln`.

However, I do feel that the implementation of these are pretty complex and is probably worth sharing. In the commit [have writeToStdinGuarded share implementation with writelnToStdinGuarded](https://github.com/flutter/flutter/pull/150120/commits/848bbc58feb4cacc12bf5a287d0b7a35add6a760), I've refactored these to accomplish this. Let me know what you think.

Reply:
lgtm